service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
```

Comment:
the immutables library ends up being exposed through the API if you do not define the Builder interface for a specific class. this means you are requiring consuming applications to include the library or they'll throw an exception at build time.

error: incompatible types: Optional<ImmutableSomeClass> cannot be converted to Optional<SomeClass>